eppo_core/src/ufc/eval_details.rs

```diff
@@ -0,0 +1,102 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::{AttributeValue, Attributes};
+
+use super::{AssignmentValue, Condition, FlagEvaluationError, Shard, Value};
+
+/// Details about feature flag evaluation.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvalFlagDetails {
+    pub flag_key: String,
+    pub subject_key: String,
+    pub subject_attributes: Attributes,
+    /// Timestamp when the flag was evaluated.
+    pub timestamp: DateTime<Utc>,
+    /// Details of configuration used for evaluation. None if configuration hasn't been fetched yet.
+    pub configuration_details: Option<ConfigurationDetails>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub result: Option<AssignmentValue>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<FlagEvaluationError>,
+    /// Key of the selected variation.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_key: Option<String>,
+    /// Value of the selected variation. Could be `None` if no variation is selected, or selected
+    /// value is absent in configuration (configuration error).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_value: Option<Value>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub allocations: Vec<EvalAllocationDetails>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ConfigurationDetails {
+    /// Timestamp when configuration was fetched by the SDK.
+    pub fetched_at: DateTime<Utc>,
+    /// Timestamp when configuration was published by the server.
+    pub published_at: DateTime<Utc>,
+    /// Environment the configuration belongs to.
+    pub environment_name: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvalAllocationDetails {
+    pub key: String,
+    pub result: EvalAllocationResult,
```

Comment:
This is the same type as `allocationEvaluationCode: AllocationEvaluationCode`.

I can rename it to:
```rust
pub evaluation_code: AllocationEvaluationCode,
```
although not sure if "evaluation code" is clearer than "evaluation result."

Reply:
In my opinion, the more naming consistency we have between SDKs, the better. It makes it easier to switch between SDKs and have the context we need without having to relearn how a new SDK works. In this case, I think both names are fine, but since the JS SDK was already released, it might be better to just use that name.